Sample Finder polish (round 2)

cnathe opened the pull request:
#### Rationale
Round 2 of polish for the Sample Finder in 22.3.

#### Related Pull Requests
* https://github.com/LabKey/sampleManagement/pull/859

#### Changes
* Add padding to loading spinner
* Choose Values panel fix for long text options and empty state
* Filter modal styling fixes for list group: remove rounded corner, reduce padding, bold when active
* Update cursor behavior and hover behavior on filter cards and edit/remove icons
* Filter modal updates to better support display on narrow screen
* Add "Go to Sample Finder" button to the dashboard insights panel

cnathe committed:
Add padding to loading spinner

cnathe committed:
Choose values display fix for empty state and long text wrapping under checkbox

cnathe committed:
missing key for horizontal bars is causing JS error/warning

cnathe committed:
update release notes

cnathe committed:
Filter modal styling fixes for list group: remove rounded corner, reduce padding, bold when active

cnathe committed:
v2.138.1-fb-sampleFinderPolish223Cory.0

cnathe committed:
Merge branch 'release22.3-SNAPSHOT' into fb_sampleFinderPolish223Cory

# Conflicts:
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
v2.138.2-fb-sampleFinderPolish223Cory.0

cnathe committed:
Update cursor behavior and hover behavior on filter cards and edit/remove icons

cnathe committed:
Filter modal updates to better support display on narrow screen

cnathe committed:
v2.138.2-fb-sampleFinderPolish223Cory.1

cnathe committed:
Merge branch 'release22.3-SNAPSHOT' into fb_sampleFinderPolish223Cory

# Conflicts:
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
v2.138.3-fb-sampleFinderPolish223Cory.0

cnathe committed:
Add "Go to Sample Finder" button to the dashboard insights panel

cnathe committed:
Jest fix for SampleFinderSection title change

cnathe committed:
v2.138.3-fb-sampleFinderPolish223Cory.1

XingY commented on packages/components/src/internal/components/search/FilterFacetedSelector.tsx:
```diff
@@ -112,7 +112,7 @@ export const FilterFacetedSelector: FC<Props> = memo(props => {
                 )}
                 <Row>
-                    <Col xs={6}>
+                    <Col xs={taggedValues?.length > 0 ? 6 : 12}>
```
Should the other Col below in the Row for tagged value also have similar conditional width check?

cnathe replied:
Done

XingY commented on packages/components/src/internal/components/search/FilterValueDisplay.tsx:
```diff
@@ -67,13 +67,13 @@ export const FilterValueDisplay: FC<FilterValueDisplayProps> = memo(props => {
-                                                    {[...values].splice(5).map(val => (
-                                                        <div>{val}</div>
+                                                    {[...values].splice(5).map((val, i) => (
+                                                        <div key={val}>{val}</div>
```
Can use either val or i as key here, but since you explicitly add i, just checking if you intended to use val.

cnathe replied:
Good catch. I had added `i` to use it for the key but then decided to just use `val` instead. Removing `i`.

labkey-susanh commented on packages/components/src/internal/components/chart/BarChartViewer.tsx:
```diff
@@ -250,21 +251,34 @@ export class BarChartViewer extends PureComponent<Props, State> {
+
+const SampleButtons: FC = () => {
+    return (
+        <RequiresPermission perms={PermissionTypes.Insert}>
```
You shouldn't need Insert permissions to see Sample Finder.

cnathe replied:
Fixed

cnathe committed:
CR feedback
- don't need insert perm for Sample Finder button on dashboard
- bootstrap div cleanup

cnathe committed:
SM dash buttons WIP

cnathe committed:
v2.138.3-fb-sampleFinderPolish223Cory.2

cnathe committed:
BarChartViewer SampleButtons jest test

cnathe committed:
update release notes date and version

cnathe committed:
yarn run lint-branch-fix

cnathe committed:
v2.138.4
